workflow/controller/controller.go

```diff
@@ -445,6 +449,10 @@ func (wfc *WorkflowController) processNextItem() bool {
 		// or was deleted, but the work queue still had an entry for it.
 		return true
 	}
+
+	wfc.workflowKeyLock.Lock(key.(string))
```

Comment:
My goal is to prevent the reconciliation loop and the archive loop from updating the workflow at the same time.

Can I do this with the throttler instead do you think?

Reply:
yes

Reply:
after investigation, the throttler does not do what is needed